airbyte-cdk/python/unit_tests/connector_builder/test_connector_builder_handler.py

```diff
@@ -32,7 +35,7 @@
                 "page_size": 10,
                 "page_size_option": {"inject_into": "request_parameter", "field_name": "page_size"},
                 "page_token_option": {"inject_into": "path", "type": "RequestPath"},
-                "pagination_strategy": {"type": "CursorPagination", "cursor_value": "{{ response._metadata.next }}"},
+                "pagination_strategy": {"type": "CursorPagination", "cursor_value": "{{ response._metadata.next }}", "page_size": 10},
```

Comment:
Without this, I would get the following error:
```
page_size_option cannot be set if the pagination strategy does not have a page_size
Traceback (most recent call last):
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/connector_builder/connector_builder_handler.py", line 64, in list_streams
    _get_http_streams(source, config)
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/connector_builder/connector_builder_handler.py", line 81, in _get_http_streams
    for stream in source.streams(config=config):
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py", line 78, in streams
    source_streams = [
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py", line 79, in <listcomp>
    self._constructor.create_component(DeclarativeStreamModel, stream_config, config)
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py", line 192, in create_component
    return self._create_component_from_model(model=declarative_component_model, config=config, **kwargs)
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py", line 198, in _create_component_from_model
    return component_constructor(model=model, config=config, **kwargs)
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py", line 435, in create_declarative_stream
    retriever = self._create_component_from_model(
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py", line 198, in _create_component_from_model
    return component_constructor(model=model, config=config, **kwargs)
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py", line 732, in create_simple_retriever
    self._create_component_from_model(model=model.paginator, config=config, url_base=url_base)
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py", line 198, in _create_component_from_model
    return component_constructor(model=model, config=config, **kwargs)
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py", line 527, in create_default_paginator
    paginator = DefaultPaginator(
  File "<string>", line 9, in __init__
  File "/Users/maxime/devel/code/airbyte/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/default_paginator.py", line 100, in __post_init__
    raise ValueError("page_size_option cannot be set if the pagination strategy does not have a page_size")
ValueError: page_size_option cannot be set if the pagination strategy does not have a page_size
```

It seems like this field is not required based on the [declarative_component_schema.yaml](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml#L174) but [this condition](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/default_paginator.py#L99) enforces that if `page_size_option` is defined, `pagination_strategy.get_page_size` is as well. It's only a problem just now because components are not initialized in `resolved_manifest`